uint8 id
uint8 activity
string message
